src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -243,14 +243,13 @@ public void DirectoryLongerThanMaxLongPath_ThrowsPathTooLongException()
 
         [ConditionalFact(nameof(LongPathsAreNotBlocked), nameof(UsingNewNormalization))]
         [ActiveIssue(20117, TargetFrameworkMonikers.Uap)]
-        [PlatformSpecific(TestPlatforms.Windows)]  // long directory path with extended syntax throws PathTooLongException
-        public void DirectoryLongerThanMaxLongPathWithExtendedSyntax_ThrowsPathTooLongException()
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void DirectoryLongerThanMaxLongPathWithExtendedSyntax_ThrowsException()
         {
             var paths = IOInputs.GetPathsLongerThanMaxLongPath(GetTestFilePath(), useExtendedSyntax: true);
-            Assert.All(paths, (path) =>
-            {
-                Assert.Throws<PathTooLongException>(() => Create(path));
-            });
+
+            // Long directory path with extended syntax throws PathTooLongException.
+            Assert.All(paths, path => { Assert.Throws<PathTooLongException>(() => Create(path)); });
```

Comment:
I presume this is all about where the directory separator falls and what the internal path gets turned into (see my blog post on the [subject](https://blogs.msdn.microsoft.com/jeremykuhne/2016/05/02/dos-to-nt-a-paths-journey/)). You can use AssertExtensions to see that it throws either PathTooLong or DirectoryNotFound on non-netfx.